src/Components/Web.JS/src/Rendering/Renderer.ts

```diff
@@ -2,19 +2,25 @@
 import { System_Object, System_String, System_Array, MethodHandle, Pointer } from '../Platform/Platform';
 import { platform } from '../Environment';
 import { RenderBatch } from './RenderBatch/RenderBatch';
-import { BrowserRenderer } from './BrowserRenderer';
+import { BrowserRenderer, EventDescriptor } from './BrowserRenderer';
 import { toLogicalElement, LogicalElement } from './LogicalElements';
+import { UIEventArgs } from './EventForDotNet';
 
 interface BrowserRendererRegistry {
   [browserRendererId: number]: BrowserRenderer;
 }
 const browserRenderers: BrowserRendererRegistry = {};
 
+let eventDispatcher = (eventDescriptor: EventDescriptor, eventArgs: UIEventArgs): Promise<void> => {
+  return DotNet.invokeMethodAsync('Microsoft.AspNetCore.Components.Web', 'DispatchEvent', eventDescriptor, JSON.stringify(eventArgs));
+};
```

Comment:
I find it a bit weird and misleading to have a "default" implementation of eventDispatcher here that gets overridden in certain cases. This sort of pattern makes it hard to interpret how the code flows when reading it statically, because you it's hard to realise it might get overridden from elsewhere.

If possible, could we not bake in a default here, but rather have both `Browser.Server.ts` and `Browser.WebAssembly.ts` provide their own implementations as peers, neither of which is privileged over the other?

Reply:
That sounds good, I wasn't sure of the pattern that we follow to plug implementation specific things, that's why I did it this way.

Is there a way to avoid the setup on the boot.*.ts? Ideally I would have this code simply perform an import and have the right implementation setup somewhere else

Reply:
> Is there a way to avoid the setup on the boot.*.ts

Not really

Reply:
done